### PR TITLE
Document some aspects of resource tracking.

### PR DESCRIPTION
**Pull request by jimblandy**
Please read carefully - these are just my inferences from reading the code, so I could have misunderstood something important. Hopefully it's at least a step forward.

### CODE REVIEW

File: wgpu-core/src/lib.rs [hunk 1]
```diff
@@ -122,10 +135,48 @@ impl MultiRefCount {
...
+    /// When the user first creates a `wgpu-core` resource, this `RefCount` is
+    /// created along with the resource's `LifeGuard`. When the user drops the
+    /// resource, we swap this out for `None`. Note that the resource may
+    /// still be held alive by other resources.
+    ///
+    /// Any `Stored<T>` value holds a clone of this `RefCount` along with the id
+    /// of a `T` resource.
     ref_count: Option<RefCount>,
+
+    /// A timestamp of the last queue submission in which the resource
```
Comment (kvark) [inline wgpu-core/src/lib.rs:170]: It's not a timestamp, technically, it's just an index.

**Review by kvark — APPROVED** [review]
Good docs, thank you!

**Comment by kvark** [comment]
formatting is angry a bit

File: wgpu-core/src/lib.rs (hunk 1 above)
Comment (jimblandy) [reply wgpu-core/src/lib.rs:170]: 'Serial number' would be an exact fit for this, but the code and comments all use 'index', so I'll go with that.

**Comment by jimblandy** [comment]
I don't think the remaining failure has anything to do with this PR.

**Comment by cwfitzgerald** [comment]
We have a spurious failure on GL multithreaded

**Pull request merged**